pkg/chunk/storage/factory.go

```diff
@@ -36,40 +33,23 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	cfg.GCPStorageConfig.RegisterFlags(f)
 	cfg.CassandraStorageConfig.RegisterFlags(f)
 
-	f.IntVar(&cfg.IndexCacheSize, "store.index-cache-size", 0, "Size of in-memory index cache, 0 to disable.")
-	f.DurationVar(&cfg.IndexCacheValidity, "store.index-cache-validity", 5*time.Minute, "Period for which entries in the index cache are valid. Should be no higher than -ingester.max-chunk-idle.")
-	cfg.memcacheClient.RegisterFlagsWithPrefix("index", f)
+	cfg.indexCache.RegisterFlagsWithPrefix("store.index-cache-read", f)
```

Comment:
Doesn't this change the flags?  Won't think break backwards compatibility?

Reply:
We can't break backward compat like this.  At least leave the old flags in with a deprecated notice.

Reply:
Done